### ISSUE TITLE
does the river use a "notimeout" cursor?

### ISSUE THREAD

**Issue by derEremit** [richardwilly98/elasticsearch-river-mongodb#542]
I'm experimenting with switching to wiredtiger db engine an just read that with wiredtiger there are serious performance issues with using notimeout cursors.

without looking at the code it seems to me that this could be what the river could be using to "hook" to the oplog

**Comment by kdkeck** [comment]
Yes, it does, but I'm not sure the performance issue with notimeout applies to the oplog, as opposed to other collections, perhaps because it's not the only flag set on the oplog cursor:

```java
        int options = Bytes.QUERYOPTION_TAILABLE | Bytes.QUERYOPTION_AWAITDATA | Bytes.QUERYOPTION_NOTIMEOUT
        // Using OPLOGREPLAY to improve performance:
        // https://jira.mongodb.org/browse/JAVA-771
                | Bytes.QUERYOPTION_OPLOGREPLAY;

        DBCursor cursor = oplogCollection.find(indexFilter).setOptions(options);
```

We migrated to wiredtiger some weeks ago, and haven't seen any problems with the river being able to follow the oplog.

**Comment by iprudhomme** [comment]
I know this is an old thread, but I was wondering if you could tell me what versions of MongoDB, ES, and the river you were using that you were successfully able to use the river?
